File: backend/app/models/task.py
import datetime
import enum
from sqlalchemy import Column, Integer, String, DateTime, Enum as SAEnum
from app.database import Base


class TaskStatus(str, enum.Enum):
    PENDING = "pending"
    DONE = "done"
    FAILED = "failed"


class TaskAction(str, enum.Enum):
    RELEASE = "release"
    PURCHASE = "purchase"


class Task(Base):
    __tablename__ = "tasks"

    id = Column(Integer, primary_key=True, autoincrement=True)
    action = Column(SAEnum(TaskAction), nullable=False)
    server_id = Column(String(50), nullable=True)
    server_name = Column(String(100), default="")
    params = Column(String(500), nullable=True)  # JSON for purchase details
    status = Column(SAEnum(TaskStatus), default=TaskStatus.PENDING)
    message = Column(String(255), nullable=True)
    created_at = Column(DateTime, default=datetime.datetime.utcnow)
    done_at = Column(DateTime, nullable=True)
    user_id = Column(Integer, nullable=False, index=True)

File: src/stores/task.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'

export interface TaskItem {
  id: number
  action: string
  serverId: string
  serverName: string
  status: 'pending' | 'done' | 'failed'
  message: string
  createdAt: string
  doneAt: string
  params?: Record<string, any>
}

export const useTaskStore = defineStore('task', () => {
  const tasks = ref<TaskItem[]>([])
  const loading = ref(false)

  async function fetchTasks() {
    loading.value = true
    try {
      const token = localStorage.getItem('token')
      const res = await fetch('/api/tasks', {
        headers: { Authorization: `Bearer ${token}` }
      })
      const d = await res.json()
      tasks.value = d.data || []
    } catch {}
    loading.value = false
  }

  async function createRelease(serverId: string, serverName: string) {
    const token = localStorage.getItem('token')
    await fetch('/api/tasks', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json', Authorization: `Bearer ${token}` },
      body: JSON.stringify({ action: 'release', serverId, serverName })
    })
    await fetchTasks()
  }

  async function executeTask(taskId: number) {
    const token = localStorage.getItem('token')
    const res = await fetch(`/api/tasks/${taskId}/execute`, {
      method: 'POST',
      headers: { Authorization: `Bearer ${token}` }
    })
    const d = await res.json()
    await fetchTasks()
    return d
  }

  async function deleteTask(taskId: number) {
    const token = localStorage.getItem('token')
    await fetch(`/api/tasks/${taskId}`, {
      method: 'DELETE',
      headers: { Authorization: `Bearer ${token}` }
    })
    await fetchTasks()
  }

  return { tasks, loading, fetchTasks, createRelease, executeTask, deleteTask }
})
